module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -282,6 +284,24 @@ namespace module::vision {
                     if (log_level <= NUClear::DEBUG || keep) {
                         balls->balls.push_back(std::move(b));
                     }
+
+                    if (horizon.vision_ground_truth.exists) {
+                        Eigen::Affine3f Hcw(horizon.Hcw.cast<float>());
+
+                        const Eigen::Vector3f rBCc = Hcw
+                                                     * Eigen::Vector3f(horizon.vision_ground_truth.rBWw.X,
+                                                                       horizon.vision_ground_truth.rBWw.Y,
+                                                                       horizon.vision_ground_truth.rBWw.Z);
+
+                        Eigen::Vector3f ball_position = axis * distance;
+                        Eigen::Vector3f ball_error;
+                        ball_error[0] = ball_position.x() - rBCc.x();
+                        ball_error[1] = ball_position.y() - rBCc.y();
+                        ball_error[2] = ball_position.z() - rBCc.z();
```

Comment:
```suggestion
                        Eigen::Vector3f ball_error = ball_position - rBCc;
```
Eigen::Vectors have overloads for their mathematical operations